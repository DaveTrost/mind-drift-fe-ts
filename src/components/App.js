import React from 'react';
import { Router, Switch } from 'react-router-dom';
import { useAuth0 } from '../react-auth0-spa';
import history from '../utils/history';
import PrivateRoute from './PrivateRoute';
import NavBar from './NavBar';
import Home from './Home';
import Profile from './Profile';
import Breathe from './Breathe';

const App = () => {
  const { loading, user } = useAuth0(); 
 
  // console.log(user.sub);
  if(loading) {
    return <div>loading ...</div>;
  }

  return (
    <div>
      <Router history={history}>
        <header>
          <NavBar />
        </header>
        <Switch>
          <PrivateRoute exact path='/' component={Home} />
          <PrivateRoute path='/profile' component={Profile} />
          <PrivateRoute path='/breathe'>
            <Breathe inhale={3} exhale={4} holdIn={2} holdOut={0} endTime={12} />
          </PrivateRoute>
        </Switch>
      </Router>
    </div>
  );
};

export default App;
