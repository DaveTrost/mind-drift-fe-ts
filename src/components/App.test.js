import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

import useAuth0Mock from '../react-auth0-spa';
jest.mock('../react-auth0-spa', () => ({}));

describe('App component', () => {
  it('renders App in Loading state', () => {
    useAuth0Mock.useAuth0 = () => ({
      loading: true
    });
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders App', () => {
    useAuth0Mock.useAuth0 = () => ({
      loading: false
    });
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
