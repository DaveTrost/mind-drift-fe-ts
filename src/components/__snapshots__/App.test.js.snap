// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<div
  className="container"
>
  <BrowserRouter>
    <Switch>
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/"
      />
      <PrivateRoute
        component={[Function]}
        path="/profile"
      />
      <PrivateRoute
        component={[Function]}
        path="/breathe"
      />
      <PrivateRoute
        component={[Function]}
        path="/settings"
      />
      <PrivateRoute
        component={[Function]}
        path="/edit-settings"
      />
    </Switch>
  </BrowserRouter>
</div>
`;

exports[`App component renders App in Loading state 1`] = `
<div>
  loading ...
</div>
`;
